mlir/include/Catalyst/IR/CatalystOps.td

```diff
@@ -77,4 +77,28 @@ def PrintOp : Catalyst_Op<"print"> {
     );
 }
 
+def CustomCallOp: Catalyst_Op<"custom_call"> {
+  let summary = "CustomCall operation";
+  let description = [{
+    Encapsulates an implementation-defined operation `call_target_name` that
+    takes `inputs` and and produces `results`.
+
+    A custom call invokes code external to Catalyst. The `inputs` are passed to the
+    external code, and the external code is expected to produce a result of the
+    given type.
+    ```
+  }];
+  let arguments = (ins
+    Variadic<AnyType>:$inputs,
+    StrAttr:$call_target_name,
+    OptionalAttr<DenseI32ArrayAttr>: $number_original_arg
+  );
+  let results = (outs Variadic<AnyType>);
+
+  let assemblyFormat = [{
+    `custom` `(`$call_target_name`)` `(` $inputs `)`
+      attr-dict `:` functional-type(operands, results)
+  }];
+}
```

Comment:
Would you add mlir tests for this custom-call op?

Reply:
I am adding some tests thanks!